dedupe/convenience.py

```diff
@@ -24,55 +24,94 @@ def consoleLabel(deduper): # pragma: no cover
     '''
 
     finished = False
+    has_previous = False
+    use_previous = False
     fields = unique(field.field
                     for field
                     in deduper.data_model.primary_fields)
 
 
-    while not finished :
-        n_match, n_distinct = (len(deduper.training_pairs['match']),
-                               len(deduper.training_pairs['distinct']))
-
-        uncertain_pairs = deduper.uncertainPairs()
-
+    buffer_len = 1 # Max number of previous operations
+    examples_buffer = []
+    record_pair_buffer = [] 
+
+    while not finished :     
+            
+        if has_previous and use_previous:
+            record_pair = record_pair_buffer[0]
+            record_pair_buffer = record_pair_buffer[1:]
+            examples_buffer = examples_buffer[1:]
+            use_previous = False
+        else:
+            uncertain_pairs = deduper.uncertainPairs()
+            assert len(uncertain_pairs) == 1 
+            record_pair = uncertain_pairs[0]
+       
+        n_match_in_buffer = sum(len(x['match']) for x in examples_buffer)
```

Comment:
if you follow my suggestion above then this would be something like. 

```n_match_in_buffer = sum(label == 'distinct' for label, pair in example_buffer)```